api/src/services/opportunities/search_opportunities.py

```diff
@@ -0,0 +1,50 @@
+from typing import Sequence, Tuple
+
+from sqlalchemy import asc, desc, select
+
+import src.adapters.db as db
+from src.constants.lookup_constants import OpportunityCategory
+from src.db.models.opportunity_models import Opportunity
+from src.pagination.pagination_models import PaginationInfo, PaginationParams
+from src.pagination.paginator import Paginator
+
+
+class SearchOpportunityParams(PaginationParams):
+    opportunity_title: str | None = None
+    is_draft: bool | None = None
+    category: OpportunityCategory | None = None
+
+
+def search_opportunities(
+    db_session: db.Session, search_opportunity_dict: dict
+) -> Tuple[Sequence[Opportunity], PaginationInfo]:
+    search_params = SearchOpportunityParams.model_validate(search_opportunity_dict)
+
+    sort_fn = asc if search_params.sorting.is_ascending else desc
+
+    stmt = (
+        select(Opportunity)
+        .order_by(sort_fn(getattr(Opportunity, search_params.sorting.order_by)))
+        .order_by()
```

Comment:
We talked about removing this extra `order_by()` before shipping, just a reminder

Reply:
Added when I merged from main